doc/NPCs.md

```diff
@@ -0,0 +1,251 @@
+}
+```
+
+### Based on the gender of the player character
+The dynamic line is chosen based on the gender of the player character, both entries must exists. Both entries are parsed as `dynamic_line`. `u` is the player character.
+
+```JSON
+{
+    "u_male": "You're a man.",
+    "u_female": "You're a woman."
+}
+```
+
+### Based on items worn by the player character
+The dynamic line is chosen based on the gender of the player character, both entries are optional, but you should make sure the NPC says something all the time. Both entries are parsed as dynamic_line. `u` is the player character. The `u_is_wearing` string should be a valid item id.
+
+```JSON
+{
+    "u_is_wearing": "fur_cat_ears",
+    "yes": "Hello, I like your ears.",
+    "no": "Hello."
+}
+```
+
+---
+
+## Responses
+A response contains at least a text, which is display to the user and "spoken" by the player character (its content has no meaning for the game) and a topic to which the dialogue will switch to. It can also have a trial object which can be used to either lie, persuade or intimidate the NPC, see `npctalk.cpp` for details. There can be different results, used either when the trial succeeds and when it fails.
+
+Format:
+```JSON
+{
+    "text": "I, the player, say to you...",
+    "condition": "...something...",
+    "trial": {
+        "type": "PERSUADE",
+        "difficulty": 10
+    },
+    "success": {
+        "topic": "TALK_DONE",
+        "effect": "...",
+        "opinion": {
+            "trust": 0,
+            "fear": 0,
+            "value": 0,
+            "anger": 0,
+            "owed": 0,
+            "favors": 0
+        }
+    },
+    "failure": {
+        "topic": "TALK_DONE"
+    }
+}
+```
+
+Alternatively a short format:
+```JSON
+{
+    "text": "I, the player, say to you...",
+    "effect": "...",
+    "topic": "TALK_WHATEVER"
+}
+```
+The short format is equivalent to (an unconditional switching of the topic, `effect` is optional):
+```JSON
+{
+    "text": "I, the player, say to you...",
+    "trial": {
+        "type": "NONE"
+    },
+    "success": {
+        "effect": "...",
+        "topic": "TALK_WHATEVER"
+    }
+}
+```
+
+
+### text
+Will be shown to the user, no further meaning.
+
+### trial
+Optional, if not defined, "NONE" is used. Otherwise one of "NONE", "LIE", "PERSUADE" or "INTIMIDATE". If "NONE" is used, the `failure` object is not read, otherwise it's mandatory.
+The `difficulty` is only required if type is not "NONE" and specifies the success chance in percent (it is however modified by various things like mutations).
+
+### success and failure
+Both objects have the same structure. `topic` defines which topic the dialogue will switch to. `opinion` is optional, if given it defines how the opinion of the NPC will change. The given values are *added* to the opinion of the NPC, they are all optional and default to 0. `effect` is a function that is executed after choosing the the response, see below.
```

Comment:
It might help to give some (brief) notes about what the opinions are.  Without having looked, I would presume that positive values for trust, fear, and anger indicate that the NPC trusts me, fears me, or is angry at me (and may go Hostile!), respectively.  I don't know what a positive value for "value" does.